### DISCUSSION TITLE
Ipopt

### DISCUSSION THREAD

**Discussion by PierreMartinon** [control-toolbox/CTDirect.jl#129] [General]
info: https://github.com/JuliaSmoothOptimizers/NLPModelsIpopt.jl/blob/main/src/NLPModelsIpopt.jl#L119
options of ipopt: https://coin-or.github.io/Ipopt/OPTIONS.html
callback: https://github.com/jump-dev/Ipopt.jl#solver-specific-callback
solution field 'stats': https://juliasmoothoptimizers.github.io/SolverCore.jl/stable/reference/#SolverCore.GenericExecutionStats

We typically use the non-default options:
- mu_strategy="adaptive" appears better than the default option (monotone) for OCP problems
- sb="yes" to remove the ipopt header at the start

Informations of IPOPT
- What is the primal_inf in IPOPT ?
- What is exactly the dual_inf (dual feasibility) in IPOPT ?
- https://coin-or.github.io/Ipopt/OUTPUT.html
- https://link.springer.com/article/10.1007/s10107-004-0559-y

**Comment by ocots** [comment]
Advanced call: see https://github.com/JuliaSmoothOptimizers/NLPModelsIpopt.jl/discussions/103

**Comment by PierreMartinon** [comment]
For linear solvers options see https://github.com/control-toolbox/CTDirect.jl/issues/69
